cve_bin_tool/VersionSignature.py

```diff
@@ -50,26 +50,35 @@ def get_mapping_data(self):
         """
         self.cursor.execute(
             'CREATE TABLE IF NOT EXISTS {}(version TEXT , sourceId TEXT PRIMARY KEY)'.format(self.table_name))
-        # fetch first entry
-        entry = self.cursor.execute('SELECT * FROM {}'.format(self.table_name)).fetchone()
+
+        self.cursor.execute(
+            'CREATE TABLE IF NOT EXISTS {} (date TEXT PRIMARY KEY)'.format('last_update_' + self.table_name)
+        )
+
         update_required = False
+
+        entry = self.cursor.execute('SELECT * FROM {}'.format('last_update_'+self.table_name))
+
         if entry is not None:
-            # Check if update is required
-            # Updates if the difference between current time and the time of latest release is greater than duration
-            cur_time = datetime.datetime.now().replace(microsecond=0)
-            time_fields = entry[1].split()
-            year_data = time_fields[0].split('-')
-            time_data = time_fields[1].split(':')
-            yy, mm, dd = int(year_data[0]), int(year_data[1]), int(year_data[2])
-            H, M, S = int(time_data[0]), int(time_data[1]), int(time_data[2])
-            last_release = datetime.datetime(yy, mm, dd, H, M, S)
-            time_elapsed = str(cur_time - last_release)
-            if int(time_elapsed.split()[0]) >= self.duration:
+            # Updates if the difference between current time and the time of last update is greater than duration
+
+            # create a date object from the string of format 'yyyy-mm-dd'
+            update_data = entry.split('-')
```

Comment:
I think Travis is failing for unrelated reasons, but when I pull a local copy to test, I get a bunch of failures on both python 3.6 and python 2.7 that look like
```
AttributeError: 'sqlite3.Cursor' object has no attribute 'split'
```
related to this line.  I think you just need to get a string out of entry because instead you're getting the cursor itself.

Reply:
Oh, part of the problem here may be that my database doesn't have anything in the table yet.  You'll need to handle that case gracefully too.

Reply:
I should have used fetchone()